chapters/annotations.tex

```diff
@@ -188,10 +188,19 @@ \subsubsection{Plot Curves}\label{plot-curves}
 end Curve;
 \end{lstlisting}
 
-The mandatory \lstinline!x! and \lstinline!y! expressions are restricted to be component references referring to a scalar variable or \lstinline!time!.
-It is an error if \lstinline!x! or \lstinline!y! does not designate a scalar variable.
+The mandatory \lstinline!x! and \lstinline!y! expressions are restricted to be result references in the form of \lstinline[language=grammar]!result-reference! in the grammar (\cref{expressions1}), referring to a scalar variable (or a derivative thereof) or \lstinline!time!.
+It is an error if \lstinline!x! or \lstinline!y! does not designate a scalar result.
+If \lstinline!x! or \lstinline!y! is a derivative, \lstinline!der(v, $n$)!, then $n$ must not exceed the maximum amount of differentiation applied to \lstinline!v! in the model.
 When the \lstinline!unit! of an \lstinline!Axis! is non-empty, it is an error if the unit of the corresponding \lstinline!x! or \lstinline!y! expression (i.e., a variable's \lstinline!unit!, or second for \lstinline!time!) is incompatible with the axis unit.
 
+The set of available result variable and derivative trajectories in a simulation result is tool-specific, but it is recommended to include at least the ones being used in figures.
+It is not standardized how a tool should handle missing results for valid result references.
```

Comment:
Personally, I don't like having this kind of statements about what isn't standardized in the specification.  It was included as a precaution against the otherwise inevitable comments about what a tool is supposed to do with any derivatives it didn't store in the result.  (Just trying to bring some extra attention to this sentence in case others have similar mixed feelings.)

Reply:
I would prefer to not add the first sentence. The reason is that it suggests that a tool must use the figure-annotations when generating the result file (so it requires using figure annotation as part of translation/simulation). In Dymola you can set up what variables to store in different ways and it wouldn't be clear how figure-annotations should override that - especially as users also want to modify that for various reasons.

For the second sentence I would prefer that missing results generate a diagnostic in some way, as currently it suggests that a tool could plot a curve with zero value or even some other variable.

That also provides a solution for the first issue, as tools may suggest how to include that variable in the result in the diagnostics (including having a button to change the setup) - without having to always include those variables.

Reply:
> That also provides a solution for the first issue, as tools may suggest how to include that variable in the result in the diagnostics (including having a button to change the setup) - without having to always include those variables.

The point of making a recommendation was to not make a forcing statement about what tools should store, and it seems important to not remove the tools' freedom to handle this in different ways.  @otronarp, maybe you too find making recommendations for what to be stored too intrusive?

Since this question is much bigger than just figures, we should also drag more people's attention to this question.  To name a few: @gkurzbach, @eshmoylova, @sjoelund, @axelmartenssonmodelon.  The upcoming web meeting would also be a good forum for discussing this.

Reply:
> The point of making a recommendation was to not make a forcing statement about what tools should store, and it seems important to not remove the tools' freedom to handle this in different ways. @otronarp, maybe you too find making recommendations for what to be stored too intrusive?

Since it was written as a recommendation I felt free to ignore it, so I saw no harm in having it. On the other hand what's the point of having the recommendation if it's going to be ignored.

Reply:
I prefer to not have recommendations that are normally ignored, and I also prefer somewhat stronger suggestions - so that missing results generate some form of warning.